src/mlpack/methods/ann/layer/repeat.hpp

```diff
@@ -0,0 +1,153 @@
+
+  /**
+   * Destroy the layers held by the model.
+   */
+  virtual ~RepeatType() { }
+
+  //! Clone the RepeatType object. This handles polymorphism correctly.
+  RepeatType* Clone() const override { return new RepeatType(*this); }
+
+  //! Copy the given RepeatType layer.
+  RepeatType(const RepeatType& other);
+  //! Take ownership of the given RepeatType layer.
+  RepeatType(RepeatType&& other) noexcept;
+  //! Copy the given RepeatType layer.
+  RepeatType& operator=(const RepeatType& other);
+  //! Take ownership of the given RepeatType layer.
+  RepeatType& operator=(RepeatType&& other) noexcept;
+
+  /**
+   * Ordinary feed forward pass of a neural network, evaluating the function
+   * f(x) by propagating the activity forward through f.
+   *
+   * @param input Input data used for evaluating the specified function.
+   * @param output Resulting output activation.
+   */
+  void Forward(const MatType& input, MatType& output) override;
+
+  /**
+   * Ordinary feed backward pass of a neural network, using 3rd-order tensors as
+   * input, calculating the function f(x) by propagating x backwards through f.
+   * Using the results from the feed forward pass.
+   *
+   * @param * (input) The input data (x) given to the forward pass.
+   * @param * (output) The propagated data (f(x)) resulting from Forward()
+   * @param gy The backpropagated error.
+   * @param g The calculated gradient.
+   */
+  void Backward(const MatType& /* input */,
+                const MatType& /* output */,
+                const MatType& gy,
+                MatType& g) override;
+
+  //! Get the repeat multiples
+  const std::vector<size_t>& Multiples() const { return multiples; }
+
+  //! Get the repeat multiples for modification
+  std::vector<size_t>& Multiples()
+  {
+    this->validOutputDimensions = false;
+    return multiples;
+  }
+
+  //! Get the interleave parameter
+  bool Interleave() const { return interleave; }
+
+  //! Get the interleave parameter for modification
+  bool& Interleave() { return interleave; }
+
+  /**
+   * @brief Computes the output dimensions of the Repeat layer.
+   *
+   * The ComputeOutputDimensions function computes the output dimensions of the
+   * Repeat layer based on the input dimensions and the repeat multiples. The
+   * output dimensions will have the same number of dimensions as the input,
+   * with all dimensions other than the one specified in the axis being the same
+   * size as the input.
+   */
+  void ComputeOutputDimensions() override;
+
+  /**
+   * Serialize the layer.
+   */
+  template<typename Archive>
+  void serialize(Archive& ar, const uint32_t /* version */);
+
+ private:
+  //! Parameter to indicate number of times to repeat along each dimension
+  std::vector<size_t> multiples;
+
+  //! Parameter to indicate whether to interleave the output
+  bool interleave;
+
+  // Cache the target indices for a single tensor for use
+  // in the forward pass.
+  std::vector<size_t> outIdxs;
```

Comment:
@akropp @shrit I have a concern that even with the use of OpenMP, the use of `std::vector<size_t>` here will be a lot slower in the forward and backward passes than the previous implementation that used Armadillo objects directly.  Do you think we could revert the relevant parts of 0253a86 to the previous implementation before merge (so, `arma::` will be used explicitly for these indices)?  We can address the Armadillo/Bandicoot compatibility issues at another time.